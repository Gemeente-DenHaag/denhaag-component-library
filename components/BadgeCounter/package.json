{
  "name": "@gemeente-denhaag/badge-counter",
  "version": "0.1.1-alpha.162",
  "description": "A Badge component",
  "bugs": "https://github.com/nl-design-system/denhaag/issues",
  "repository": {
    "type": "git",
    "url": "https://github.com/nl-design-system/denhaag.git",
    "directory": "components/BadgeCounter"
  },
  "license": "EUPL-1.2",
  "author": "Municipality of The Hague",
  "exports": "./dist/index.js",
  "main": "dist/index.js",
  "module": "dist/index.js",
  "types": "dist/index.d.ts",
  "files": [
    "dist"
  ],
  "scripts": {
    "build": "rollup -c ../../rollup.config.js",
    "clean": "yarn rimraf dist tsconfig.tsbuildinfo"
  },
  "dependencies": {
    "@gemeente-denhaag/dotindicator": "0.1.1-alpha.162"
  },
  "peerDependencies": {
    "react": "^17.0.0"
  },
  "gitHead": "dcf72a9b79266c1ebede35aff4a02dd9121a980f"
}
