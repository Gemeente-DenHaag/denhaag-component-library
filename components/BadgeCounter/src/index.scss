@import "@utrecht/components/badge-counter/bem";

.denhaag-badge-counter {
  --denhaag-dot-indicator-size: 8px;
}

.denhaag-badge-counter__counter {
  @extend .utrecht-badge-counter;

  font-family: var(--utrecht-badge-counter-font-family);
  font-size: var(--utrecht-badge-counter-font-size);
}
