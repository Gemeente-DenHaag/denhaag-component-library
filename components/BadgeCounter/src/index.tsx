import React, { HTMLAttributes } from 'react';
import { DotIndicator } from '@gemeente-denhaag/dotindicator';
import './index.scss';
import clsx from 'clsx';

export type BadgeCounterProps = HTMLAttributes<HTMLDivElement>;

/**
 * A counter badge notifies a user of a specific amount of updates.
 * @param props The properties of a Badge component.
 * @constructor Constructs an instance of Badge.
 */
export const BadgeCounter: React.FC<BadgeCounterProps> = (props: BadgeCounterProps) => {
  const rootClassNames = clsx('denhaag-badge-counter', props.className);
  return (
    <div id={props.id} className={rootClassNames}>
      <DotIndicator overlap={'rectangle'}>
        <div className={'denhaag-badge-counter__counter'}>{props.children}</div>
      </DotIndicator>
    </div>
  );
};

export default BadgeCounter;
