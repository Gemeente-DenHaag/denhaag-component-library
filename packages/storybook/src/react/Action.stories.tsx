import React from 'react';
import { Meta, StoryObj } from '@storybook/react';
import { ActionSingle, ActionSingleProps } from '@gemeente-denhaag/action';
import { FormatDateLabels } from '@gemeente-denhaag/utils';

import readme from '../../../../components/Action/README.md';

const labels: FormatDateLabels = {
  today: 'vandaag',
  yesterday: 'gisteren',
  before: 'vóór',
  approachingDeadline: (daysDifference: number) => {
    if (daysDifference === 1) {
      return `nog ${daysDifference} dag`;
    }
    return `nog ${daysDifference} dagen`;
  },
};

const exampleArgs: ActionSingleProps = {
  children: <strong>Taak</strong>,
  link: '#example',
  labels,
};

const meta = {
  id: 'react-action',
  title: 'React/Actions/Action',
  component: ActionSingle,
  args: exampleArgs,
  tags: ['autodocs'],
  parameters: {
    chromatic: { viewports: [1768, 1280, 768, 360] },
    docs: {
      description: {
        component: readme,
      },
    },
  },
} as Meta<typeof ActionSingle>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {};

export const Hover: Story = {
  args: { ...Default.args, className: 'denhaag-action--hover' },
};

export const Focus: Story = {
  args: { ...Default.args, className: 'denhaag-action--focus' },
};

export const WithDate: Story = {
  args: { ...Default.args, dateTime: '2023-09-28T19:47:36.593Z' },
};

export const WithDateRelative: Story = {
  args: { ...Default.args, dateTime: '2023-10-02T19:47:36.593Z', now: '2023-09-28T19:47:36.593Z', relativeDate: true },
};

export const WithDateWarning: Story = {
  args: { ...Default.args, dateTime: '2023-09-30T19:47:36.593Z', now: '2023-09-28T19:47:36.593Z', relativeDate: true },
};

// ### Multiple - hover

// <Canvas>
//   <Story name="Multiple - hover">
//     {() => (
//       <Action
//         actions={
//           <ButtonLink href="#example" className="utrecht-button-link--hover">
//             Actie
//           </ButtonLink>
//         }
//       >
//         <strong>Taak</strong>
//       </Action>
//     )}
//   </Story>
// </Canvas>

// ### Multiple - focus

// <Canvas>
//   <Story name="Multiple - focus">
//     {() => (
//       <Action
//         actions={
//           <ButtonLink href="#example" className="utrecht-button-link--focus">
//             Actie
//           </ButtonLink>
//         }
//       >
//         <strong>Taak</strong>
//       </Action>
//     )}
//   </Story>
// </Canvas>

// ### Multiple - with date

// <Canvas>
//   <Story name="Multiple - with date">
//     {() => (
//       <Action
//         dateTime={"2023-09-28T19:47:36.593Z"}
//         now={"2023-09-28T19:47:36.593Z"}
//         labels={labels}
//         actions={<ButtonLink href="#example">Actie</ButtonLink>}
//       >
//         <strong>Taak</strong>
//       </Action>
//     )}
//   </Story>
// </Canvas>

// ### Multiple - with date - relative

// <Canvas>
//   <Story name="Multiple - with date - relative">
//     {() => (
//       <Action
//         relativeDate
//         dateTime={"2023-10-02T19:47:36.593Z"}
//         now={"2023-09-28T19:47:36.593Z"}
//         labels={labels}
//         actions={<ButtonLink href="#example">Actie</ButtonLink>}
//       >
//         <strong>Taak</strong>
//       </Action>
//     )}
//   </Story>
// </Canvas>

// ### Multiple - with warning date - relative

// <Canvas>
//   <Story name="Multiple - with warning date - relative">
//     {() => (
//       <Action
//         relativeDate
//         dateTime={"2023-09-30T19:47:36.593Z"}
//         now={"2023-09-28T19:47:36.593Z"}
//         labels={labels}
//         actions={<ButtonLink href="#example">Actie</ButtonLink>}
//       >
//         <strong>Taak</strong>
//       </Action>
//     )}
//   </Story>
// </Canvas>
