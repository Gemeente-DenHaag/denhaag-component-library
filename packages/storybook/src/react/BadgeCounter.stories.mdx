import { ArgsTable, Canvas, Description, Meta, Story } from "@storybook/blocks";

import { BadgeCounter } from "@gemeente-denhaag/components-react";
import pkg from "../../../../components/Button/package.json";
import readme from "../../../../components/Button/README.md";

<Meta
  title="React/Data Display/Badge Counter"
  component={BadgeCounter}
  parameters={{
    componentSubtitle: `${pkg.name} - ${pkg.version}`,
    docs: { source: { type: "dynamic" } },
    status: {
      type: "ALPHA",
    },
  }}
/>

<Description markdown={readme} />

## Stories

### Default

<Canvas>
  <Story name="Default">{(args) => <BadgeCounter {...args}>13</BadgeCounter>}</Story>
</Canvas>

<ArgsTable of={BadgeCounter} />
