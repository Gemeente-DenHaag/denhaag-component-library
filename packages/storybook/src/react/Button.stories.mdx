import { ArgsTable, Canvas, Description, Meta, Story } from "@storybook/blocks";

import { ArrowLeftIcon, ArrowRightIcon } from "@gemeente-denhaag/icons";

import { Button } from "../index";
import pkg from "../../package.json";
import readme from "../../README.md";

<Meta
  title="React/Input/Button"
  component={Button}
  parameters={{
    componentSubtitle: `${pkg.name} - ${pkg.version}`,
    docs: { source: { type: "dynamic" } },
    status: {
      type: "ALPHA",
    },
  }}
/>

<Description markdown={readme} />

## Stories

### Default

<Canvas>
  <Story name="Default" args={{}}>
    {(args) => <Button {...args}>Button</Button>}
  </Story>
</Canvas>

<ArgsTable of={Button} />

### Primary Action

<Canvas>
  <Story
    name="Primary Action"
    args={{
      variant: "primary-action",
    }}
  >
    {(args) => <Button {...args}>Button</Button>}
  </Story>
</Canvas>

### Secondary Action

<Canvas>
  <Story
    name="Secondary Action"
    args={{
      variant: "secondary-action",
    }}
  >
    {(args) => <Button {...args}>Button</Button>}
  </Story>
</Canvas>

### Icon Left

<Canvas>
  <Story
    name="Icon Left"
    args={{
      icon: <ArrowLeftIcon />,
    }}
  >
    {(args) => <Button {...args}>Button</Button>}
  </Story>
</Canvas>

### Icon Right

<Canvas>
  <Story
    name="Icon Right"
    args={{
      variant: "secondary-action",
      icon: <ArrowRightIcon />,
      iconAlign: "end",
    }}
  >
    {(args) => <Button {...args}>Button</Button>}
  </Story>
</Canvas>

### Medium

<Canvas>
  <Story
    name="Medium"
    args={{
      size: "medium",
    }}
  >
    {(args) => <Button {...args}>Button</Button>}
  </Story>
</Canvas>

### Large

<Canvas>
  <Story
    name="Large"
    args={{
      size: "large",
    }}
  >
    {(args) => <Button {...args}>Button</Button>}
  </Story>
</Canvas>

### Disabled

<Canvas>
  <Story
    name="Disabled"
    args={{
      disabled: true,
    }}
  >
    {(args) => <Button {...args}>Button</Button>}
  </Story>
</Canvas>
