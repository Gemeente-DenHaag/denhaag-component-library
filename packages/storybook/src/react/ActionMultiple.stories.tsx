import React from 'react';
import { Meta, StoryObj } from '@storybook/react';
import { ActionMulti } from '@gemeente-denhaag/action';
import { ButtonLink } from '@utrecht/component-library-react';

import readme from '../../../../components/Action/README.md';

const labels = {
  today: 'vandaag',
  yesterday: 'gisteren',
  before: 'vóór',
  approachingDeadline: (daysDifference: number) => {
    if (daysDifference === 1) {
      return `nog ${daysDifference} dag`;
    }
    return `nog ${daysDifference} dagen`;
  },
};

const exampleArgs = {
  labels,
  children: <strong>Taak</strong>,
  actions: <ButtonLink href="#example">Actie</ButtonLink>,
};

const meta = {
  id: 'react-actions-action-multiple',
  title: 'React/Actions/Action Multiple',
  component: ActionMulti,
  args: exampleArgs,
  tags: ['autodocs'],
  parameters: {
    chromatic: { viewports: [1768, 1280, 768, 360] },
    docs: {
      description: {
        component: readme,
      },
    },
  },
} as Meta<typeof ActionMulti>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {};

export const Hover: Story = {
  args: {
    ...Default.args,
    actions: (
      <ButtonLink href="#example" className="utrecht-button-link--hover">
        Actie
      </ButtonLink>
    ),
  },
};

export const Focus: Story = {
  args: {
    ...Default.args,
    actions: (
      <ButtonLink href="#example" className="utrecht-button-link--focus">
        Actie
      </ButtonLink>
    ),
  },
};

export const WithDate: Story = {
  args: { ...Default.args, dateTime: '2023-09-28T19:47:36.593Z' },
};

export const WithDateRelative: Story = {
  args: { ...Default.args, dateTime: '2023-10-02T19:47:36.593Z', now: '2023-09-28T19:47:36.593Z', relativeDate: true },
};

export const WithDateWarning: Story = {
  args: { ...Default.args, dateTime: '2023-09-30T19:47:36.593Z', now: '2023-09-28T19:47:36.593Z', relativeDate: true },
};
