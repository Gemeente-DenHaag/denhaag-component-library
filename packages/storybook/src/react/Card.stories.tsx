import React from 'react';
import { Meta, Story } from '@storybook/react';
import Card, { CardProps } from '../index';
import pkg from '../../package.json';
import SubjectCard, { SubjectCardProps } from '../Card/SubjectCard';
import CaseCard, { CaseCardProps } from '../Card/CaseCard';

export default {
  title: 'React/Cards/Card',
  parameters: {
    componentSubtitle: `${pkg.name} - ${pkg.version}`,
    status: {
      type: 'ALPHA',
    },
  },
  component: Card,
} as Meta;

const Template: Story<SubjectCardProps> = (args: CardProps) => <SubjectCard {...args} />;

const CaseTemplate: Story<CaseCardProps> = (args: CaseCardProps) => <CaseCard {...args} />;

/**
 * Implementation of Card
 */
export const Default: Story<SubjectCardProps> = Template.bind({});
Default.args = {
  title: 'Shrimp and Chorizo Paella',
  subTitle: 'This impressive paella is a perfect party dish and a fun meal to cook.',
  date: new Date('2020-01-21'),
  href: '#',
};

/**
 * Default variant of Card component without date
 */
export const CardWithoutDate: Story<SubjectCardProps> = Template.bind({});
CardWithoutDate.args = {
  title: Default.args.title,
  subTitle: Default.args.subTitle,
  href: '#',
};

/**
 * Case active variant of Card component
 */
export const CaseActive: Story<CaseCardProps> = CaseTemplate.bind({});
CaseActive.args = {
  title: Default.args.title,
  subTitle: Default.args.subTitle,
  date: Default.args.date,
  href: '#',
};

/**
 * Case variant of Card component
 */
export const CaseInactive: Story<CaseCardProps> = CaseTemplate.bind({});
CaseInactive.args = {
  title: Default.args.title,
  subTitle: Default.args.subTitle,
  date: Default.args.date,
  active: false,
  href: '#',
};
