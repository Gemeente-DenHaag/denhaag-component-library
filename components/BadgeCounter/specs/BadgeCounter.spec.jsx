import * as React from 'react';
import BadgeCounter from '../dist';

describe(`BadgeCounter tests`, () => {
  it('a11y and snapshots', () => {
    cy.snapshots(<BadgeCounter>13</BadgeCounter>, {});
  });
});
