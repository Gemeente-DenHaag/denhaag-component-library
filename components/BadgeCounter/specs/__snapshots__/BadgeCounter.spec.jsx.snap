exports[`BadgeCounter tests > a11y and snapshots #0`] = `
<div id="wrapper" style="display: inline-block;">
  <div style="padding: 10px;">
    <div class="denhaag-badge-counter">
      <span
        class="denhaag-dot-indicator denhaag-dot-indicator--overlap-rectangle"
        ><div class="denhaag-badge-counter__counter">13</div>
        <span class="denhaag-dot-indicator__dot"></span
      ></span>
    </div>
  </div>
</div>
`;
